src/puppetlabs/puppetdb/config.clj

```diff
@@ -245,7 +245,10 @@
   "Checks that `vardir` is specified, exists, and is writeable, throwing
   appropriate exceptions if any condition is unmet."
   [config]
-  (let [vardir (io/file (get-in config [:global :vardir]))]
+  (let [vardir (some-> (get-in config [:global :vardir])
+                       io/file
+                       fs/absolute-path
+                       fs/file)]
```

Comment:
You are probably correct about both of the above, thanks.  Will fix, assuming you guys are OK with the behavior change.